src/datasources/cache/cache.first.data.source.ts

```diff
@@ -40,15 +41,22 @@ export class CacheFirstDataSource {
    * @param args.networkRequest - the HTTP request to be used if there is a cache miss
    * @param args.expireTimeSeconds - the time to live in seconds for the payload behind {@link CacheDir}
    * @param args.notFoundExpireTimeSeconds - the time to live in seconds for the error when the item is not found
+   * @param args.isRateLimited - indicates if the call is rate-limited. If so, and the data is not already cached, an error will be thrown
    */
   async get<T>(args: {
     cacheDir: CacheDir;
     url: string;
     notFoundExpireTimeSeconds: number;
     networkRequest?: NetworkRequest;
     expireTimeSeconds?: number;
+    isRateLimited?: boolean;
```

Comment:
This argument is added to inform the function the data should be retrieved exclusively from the cache. 

Another option, besides adding this parameter, would be to inject both `ICacheService` and `INetworkService` into the caller (`ZerionBalancesApi`) and perform the logic there, but I thought having this parameter could be valuable for other classes in the future. 

Having that said, I haven't a strong opinion about the alternatives. Feedback is welcomed :)

Reply:
```
We need not inject anything. We could throw if the rate limit was hit
```

That was something I also thought at first glance, but it wasn't so simple. We need to avoid throwing if the data is available in the cache. Otherwise, we would be blocking all the `getBalances` calls, not only the ones that need a call to Zerion.

Reply:
I see. I would still lean towards moving this out of the data source. If that's difficult we can keep it as is though, but I'd tweak the name, e.g. `shouldImmediatelyReturn`.

Reply:
I've removed the parameter as you suggested, and now all the logic is performed within `ZerionBalancesApi`. If more data source APIs need to be rate-limited we can think in including back the parameter :) 

[23d9d5e](https://github.com/safe-global/safe-client-gateway/pull/1228/commits/23d9d5e6730da71544fe9b883c2cc35d8fe55fa1)